starter-code/index.js

```diff
@@ -1,10 +1,89 @@
-var SortedList = function() {};
-
-SortedList.prototype.add     = function(item) {}
-SortedList.prototype.get     = function(pos) {}
-SortedList.prototype.max     = function() {}
-SortedList.prototype.min     = function() {}
-SortedList.prototype.average = function() {}
-SortedList.prototype.sum     = function() {}
+let SortedList = function() {
+    this.items = [];
+    this.length = this.items.length;
+};
+
+SortedList.prototype.add = function(item) {
+    
+    this.items.push(item)
+    this.items.sort(function(i1, i2){
+        return i1-i2});
+    
+    this.length++;
+}
+SortedList.prototype.get = function(pos) {
+    if(pos > this.items.length){
+        return this.OutOfBounds();
+    }
+    return this.items[pos-1];
+}
+SortedList.prototype.max = function() {
+    if (this.items === 0) {
+        return this.EmptyList();
+    } else {
+        var max = this.items[0];
+        this.items.forEach(function(e) {
+            if(e > max){
+                max = e;
+            }
+        }.bind(this));
+    }
+    return max; 
+}
+SortedList.prototype.min = function() {
+    if (this.items.length === 0) {
+        return this.EmptyList();
+    } else {
+        var min = this.items[0];
+        this.items.forEach(function(e) {
+            if(e < min){
+                min = e;
+            }
+        }.bind(this));
+    }
+    return min;
+}
+SortedList.prototype.average = function() {
+    if (this.items.length === 0) {
+        return this.EmptySortedList();
+    } else {
+        return this.items.reduce(function(avg, e, i, arr) {
+            var average = 0;
+            if (i === this.items.length - 1) {
+                return average = ((avg + e) / this.items.length);
+            } else {
+                return average = avg + e;
+            }
+        }.bind(this), 0);
+        return average;
+    }
+}
+
+
+SortedList.prototype.sum = function() {
+    if (this.items.length === 0) {
+        return this.EmptySortedList();
+    } else {
+        return this.items.reduce(function(sum, e) {
+          return sum += e;
+        }.bind(this), 0);
+    }
+}
+
+
+
+
+SortedList.prototype.OutOfBounds = function(){
+    return "Sorry, the position is out of bounds";
+}
+
+SortedList.prototype.EmptyList = function() {
+    return "The list is empty";
+
+}
+
+SortedList.prototype.EmptySortedList = function() {
+    return "The sorted list is empty";
+}
```

Comment:
Con estas tres funciones, estáis creando sobreabstracción. La intención es buena, pero si solo es para imprimir un mensaje estáis escribiendo más que si pusiérais un `console.log()` donde llamáis a cada función. Por no decir que teniendo en cuenta que las tres devuelven un mensaje, podríais haber unificado dichos tres mensajes en una sola función que devuelva diferentes mensajes según el caso.